Plugin/Model/Order/ItemRepositoryPlugin.php

```diff
@@ -0,0 +1,179 @@
+     */
+    private $serializer;
+
+    /**
+     * ItemRepositoryPlugin constructor.
+     *
+     * @param OrderItemExtensionFactory $orderItemExtensionFactory
+     * @param GetSourceSelectionResultFromOrder $sourceSelectionResult
+     * @param GetInventoryRequestFromOrder $getInventoryRequestFromOrder
+     * @param GetDefaultSourceSelectionAlgorithmCodeInterface $getDefaultSourceSelectionAlgorithmCode
+     * @param SourceSelectionServiceInterface $sourceSelectionService
+     * @param SerializerInterface $serializer
+     */
+    public function __construct(
+        OrderItemExtensionFactory $orderItemExtensionFactory,
+        GetSourceSelectionResultFromOrder $sourceSelectionResult,
+        GetInventoryRequestFromOrder $getInventoryRequestFromOrder,
+        GetDefaultSourceSelectionAlgorithmCodeInterface $getDefaultSourceSelectionAlgorithmCode,
+        SourceSelectionServiceInterface $sourceSelectionService,
+        SerializerInterface $serializer
+    ) {
+        $this->orderItemExtensionFactory = $orderItemExtensionFactory;
+        $this->sourceSelectionResult = $sourceSelectionResult;
+        $this->getInventoryRequestFromOrder = $getInventoryRequestFromOrder;
+        $this->getDefaultSourceSelectionAlgorithmCode = $getDefaultSourceSelectionAlgorithmCode;
+        $this->sourceSelectionService = $sourceSelectionService;
+        $this->serializer = $serializer;
+    }
+
+    /**
+     * Set additional attributes when requesting a single order item
+     *
+     * @param ItemRepository  $subject
+     * @param OrderItemInterface|Item $result
+     * @return OrderItemInterface|Item
+     */
+    public function afterGet(ItemRepository $subject, OrderItemInterface $result)
+    {
+        return $this->applyExtensionAttributesToOrderItem($result, $result);
+    }
+
+    /**
+     * Set additional attributes when requesting a list of order items
+     *
+     * @param ItemRepository $subject
+     * @param OrderItemSearchResultInterface $result
+     * @return OrderItemSearchResultInterface
+     */
+    public function afterGetList(ItemRepository $subject, OrderItemSearchResultInterface $result)
+    {
+        foreach ($result->getItems() as $item) {
+            $this->applyExtensionAttributesToOrderItem($item, $result->getItems());
+        }
+
+        return $result;
+    }
+
+    /**
+     * Apply extension attributes to order item
+     *
+     * @param OrderItemInterface $orderItem
+     * @param array $allItems
+     * @return OrderItemInterface
+     */
+    private function applyExtensionAttributesToOrderItem(OrderItemInterface $orderItem, array $allItems): OrderItemInterface
+    {
+        if (!$extensionAttributes = $orderItem->getExtensionAttributes()) {
+            $extensionAttributes = $this->orderItemExtensionFactory->create();
+        }
+
+        $extensionAttributes->setSources(
+            $this->serializer->serialize(
+                $this->getOrderItemSources($orderItem, $allItems)
```

Comment:
Are we certain we want to store this data as a JSON blob? We could create a class/interface to set against the order item extension attribute for this data that has the `source_code` and `qty` properties and then any other functionality can take that class and do whatever with it (serialise etc)